### DISCUSSION TITLE
Windows release with ANGLE support

### DISCUSSION THREAD

**Discussion by etnav** [maplibre/maplibre-native#890] [General]
Hi, 

would it be possible to have a separate windows release with ANGLE support?

Currently, when installing `Maplibre GL Native` via NPM (which is great btw), this installs the release without ANGLE support. My current workaround is to build `Maplibre GL Native` manually and then replace the `mbgl.node` with the newly built `mbgl.node` where ANGLE support is enabled. Quiet the overhead in my opinion.

Interested to hear about why the default is to build without ANGLE support, since chromium (and it doesn't stop there) uses ANGLE.

@tdcosta100 Tagging you, since you did a great job with the windows build support and maybe you got an answer for me 😄

**Comment by acalcutt** [comment]
Did you have to make any changes to support ANGLE or was it just alternate build commands? Are there any downsides to adding support?

Right now it is tough to support multiple binaries for one platform, due to the way node-pre-gyp downloads a binary file based on a [platform/arch/node-abi string](https://github.com/maplibre/maplibre-gl-native/blob/main/package.json#L73) . In the release workflows, it mostly builds the release with the default options.

In testing, I have found it should be possible to supply node-pre-gyp a mirror where alternate binaries would be stored.
`npm i --mbgl_binary_host_mirror=https://foo.lan/maplbre-native/alternate_version_mirror/`
you can see some example of this in the node-pre-gyp README https://github.com/mapbox/node-pre-gyp#download-binary-files-from-a-mirror .

**Reply by etnav** [reply]
No changes needed since the CMake workflow already allows for the `MBGL_WITH_EGL` flag to be used and it builds with ANGLE. Already suspected that multiple releases for a single platform would be tough 😢

**Comment by tdcosta100** [comment]
Hi, @etnav. I think ANGLE support would be bring as build option. Binaries other than the default build would be a separate project. Maybe when the project is split by platform, we could build releases for all variants (like EGL and OSMesa) we can bring this by default, but for now, I think it would be new packages, like @maplibre/maplibre-gl-native-egl, @maplibre/maplibre-gl-native-osmesa @maplibre/maplibre-gl-native-angle, etc. Or not hahaha. Just throwing ideas.
